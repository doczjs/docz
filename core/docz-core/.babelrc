{
  "plugins": ["lodash"],
  "presets": [
    "@babel/preset-typescript",
    [
      "@babel/preset-env",
      {
        "targets": {
          "node": "current"
        }
      }
    ]
  ]
}
