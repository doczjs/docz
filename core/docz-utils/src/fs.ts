import * as path from 'path'
import * as fs from 'fs-extra'
import { compile } from 'art-template'

import { format } from './format'

export const touch = (file: string, raw: string, formatFiles: boolean) =>
  new Promise(async (resolve, reject) => {
    const content = /jsx?$/.test(path.extname(file))
      ? await format(raw, formatFiles)
      : raw
    const stream = fs.createWriteStream(file)

    stream.write(content, 'utf-8')
    stream.on('finish', () => resolve())
    stream.on('error', err => reject(err))
    stream.end()
  })

export const compiled = (file: string, opts: Record<string, any> = {}) => {
  const raw = fs.readFileSync(file, { encoding: 'utf-8' })
  return compile(raw, opts)
}
