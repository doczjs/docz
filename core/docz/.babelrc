{
  "plugins": ["@babel/plugin-syntax-dynamic-import", "lodash"]
}
