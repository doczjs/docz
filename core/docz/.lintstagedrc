{
  "*.{ts,tsx}": ["npm run fix", "git add"]
}
