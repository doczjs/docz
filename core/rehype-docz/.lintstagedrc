{
  "*.{ts,tsx}": ["yarn fix", "git add"]
}
