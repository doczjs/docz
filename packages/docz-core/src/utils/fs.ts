import * as path from 'path'
import * as fs from 'fs-extra'
import { compile } from 'art-template'
import { format } from 'docz-utils'

export const touch = (file: string, raw: string) =>
  new Promise(async (resolve, reject) => {
    const content = /jsx?$/.test(path.extname(file)) ? await format(raw) : raw
    const stream = fs.createWriteStream(file)

    stream.write(content, 'utf-8')
    stream.on('finish', () => resolve())
    stream.on('error', err => reject(err))
    stream.end()
  })

export const read = (file: string): Promise<string> =>
  new Promise<string>((resolve, reject) => {
    let data = ''
    const stream = fs.createReadStream(file, { encoding: 'utf-8' })

    stream.on('data', chunk => (data += chunk))
    stream.on('end', () => resolve(data))
    stream.on('error', err => reject(err))
  })

export const readIfExist = async (file: string): Promise<string | null> => {
  const exist = await fs.pathExists(file)
  return exist ? read(file) : Promise.resolve(null)
}

export const compiled = async (
  file: string,
  opts: Record<string, any> = {}
): Promise<(args: any) => string> =>
  read(file)
    .then(data => compile(data, opts))
    .catch(err => err)
