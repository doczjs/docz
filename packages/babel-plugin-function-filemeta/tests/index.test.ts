import { transformSync } from '@babel/core'
import * as fs from 'fs'
import * as path from 'path'
import plugin from '../src'

const fixture = path.resolve('./tests/fixtures/example.js')

const code = fs.readFileSync(fixture).toString()

it('works', () => {
  const result = transformSync(code, {
    plugins: [plugin],
    filename: fixture,
  })

  expect(result!.code).toMatchSnapshot()
})
