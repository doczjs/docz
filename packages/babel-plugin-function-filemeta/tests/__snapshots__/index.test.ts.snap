// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.getOther = getOther;
exports.default = exports.Abcd = exports.bar = void 0;

const foo = () => 'foo';

foo.__filemeta = {
  name: \\"foo\\",
  filename: \\"/Volumes/Projects/dev/docz/docz/packages/babel-plugin-function-filemeta/tests/fixtures/example.js\\"
};

const bar = () => 'bar';

exports.bar = bar;
bar.__filemeta = {
  name: \\"bar\\",
  filename: \\"/Volumes/Projects/dev/docz/docz/packages/babel-plugin-function-filemeta/tests/fixtures/example.js\\"
};
const obj = {
  foo: () => 'foo'
  /**
   * Some description
   */

};

function get(object, path) {
  return object[path];
}

get.__filemeta = {
  name: \\"get\\",
  filename: \\"/Volumes/Projects/dev/docz/docz/packages/babel-plugin-function-filemeta/tests/fixtures/example.js\\"
};

function getOther(object, path) {
  return object[path];
}

getOther.__filemeta = {
  name: \\"getOther\\",
  filename: \\"/Volumes/Projects/dev/docz/docz/packages/babel-plugin-function-filemeta/tests/fixtures/example.js\\"
};
const obj2 = {
  get: () => null
  /**
   * Some description
   */

};

class Abc {
  /**
   * Some description
   */
  method() {
    return null;
  }

}

Abc.__filemeta = {
  name: \\"Abc\\",
  filename: \\"/Volumes/Projects/dev/docz/docz/packages/babel-plugin-function-filemeta/tests/fixtures/example.js\\"
};

class Abcd {
  /**
   * Some description
   */
  method() {
    return null;
  }

}

exports.Abcd = Abcd;
Abcd.__filemeta = {
  name: \\"Abcd\\",
  filename: \\"/Volumes/Projects/dev/docz/docz/packages/babel-plugin-function-filemeta/tests/fixtures/example.js\\"
};
const obj3 = {
  method: class Abcd {
    method() {
      return null;
    }

  }
};

const log = fn => fn(1);

log.__filemeta = {
  name: \\"log\\",
  filename: \\"/Volumes/Projects/dev/docz/docz/packages/babel-plugin-function-filemeta/tests/fixtures/example.js\\"
};
log(id => console.log(id));

var _default = () => 'other';

exports.default = _default;"
`;
